Reject if subaccount is set

leongb opened the pull request.

jleni reviewed (changes requested):
Do we still need to update the test vectors to go with this?

leongb commented:
I updated the pb part as we could not handle the subaccount (it freezes), which means that would also be the current behaviour. I modified it so we return an error instead. I also changed the way we handle serialized_id's, it should be good now
